### PR TITLE
New README.md file for rule index

### PR DESCRIPTION
**Pull request by laurentlb**

### CODE REVIEW

File: rules/mandatory_provider/BUILD [hunk 1]
```diff
@@ -0,0 +1,19 @@
+load(":sum.bzl", "sum")
+
+sum(
+    name = "n",
+    deps = [
+        "n2",
```
Comment (vladmos) [inline rules/mandatory_provider/BUILD]: ":n2" to avoid confusion (here and below)

File: rules/optional_provider/BUILD [hunk 2]
```diff
@@ -0,0 +1,19 @@
+load(":sum.bzl", "sum")
+
+sum(
+    name = "n",
+    deps = [
+        "n2",
```
Comment (vladmos) [inline rules/optional_provider/BUILD]: ":n2" to avoid confusion (here and below)

File: rules/runfiles/execute.bzl [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      runfiles=ctx.runfiles(files=ctx.files.data)
+  )]
+
+execute = rule(
+  implementation=_impl,
+  executable=True,
+  attrs={
+      "command": attr.string(),
+      "data": attr.label_list(allow_files=True),
+      },
```
Comment (vladmos) [inline rules/runfiles/execute.bzl]: nit: remove 4 spaces

**Comment by laurentlb** [comment]
Updated

**Pull request merged**